.NET/Microsoft.Recognizers.Text.DataTypes.TimexExpression/TimexResolver.cs

```diff
@@ -265,24 +309,88 @@ private static DateObject FirstDateOfWeek(int year, int weekOfYear, System.Globa
             return firstWeekDay.AddDays(weekOfYear * 7);
         }
 
-        private static Tuple<string, string> MonthWeekDateRange(int year, int month, int weekOfYear)
+        private static Tuple<string, string> MonthWeekDateRange(int year, int month, int weekOfMonth)
         {
-            var dateInWeek = new DateObject(year, month, 1 + ((weekOfYear - 1) * 7));
-            if (dateInWeek.DayOfWeek == DayOfWeek.Sunday)
+            var start = GenerateMonthWeekDateStart(year, month, weekOfMonth);
+            var end = start.AddDays(7);
+
+            return new Tuple<string, string>(
+                TimexValue.DateValue(new TimexProperty { Year = start.Year, Month = start.Month, DayOfMonth = start.Day }),
+                TimexValue.DateValue(new TimexProperty { Year = end.Year, Month = end.Month, DayOfMonth = end.Day }));
+        }
+
+        private static DateObject GenerateMonthWeekDateStart(int year, int month, int weekOfMonth)
+        {
+            var dateInWeek = new DateObject(year, month, 1 + ((weekOfMonth - 1) * 7));
+
+            // Align the date of the week according to Thursday, base on ISO 8601, https://en.wikipedia.org/wiki/ISO_8601
+            if (dateInWeek.DayOfWeek > DayOfWeek.Thursday)
```

Comment:
Same in recognizer, refer to https://github.com/microsoft/Recognizers-Text/blob/master/.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs#L1768